CMakeLists.txt

```diff
@@ -183,3 +183,15 @@ endif()
 if(IDF_TARGET STREQUAL "esp32s2")
 target_compile_options(${COMPONENT_TARGET} PUBLIC -DARDUINO=10812 -DARDUINO_ESP32S2_DEV -DARDUINO_ARCH_ESP32 -DARDUINO_BOARD="ESP32S2_DEV" -DARDUINO_VARIANT="esp32s2" -DESP32)
 endif()
+
+if(CONFIG_AUTOSTART_ARDUINO)
+    # in autostart mode, arduino-esp32 contains app_main() function and needs to
+    # reference setup() and loop() in the main component. If we add main
+    # component to priv_requires then we create a large circular dependency
+    # (arduino-esp32 -> main -> arduino-esp32) and can get linker errors, so
+    # instead we add setup() and loop() to the undefined symbols list so the
+    # linker will always include them.
+    #
+    # (As they are C++ symbol, we need to add the C++ mangled names.)
+    target_link_libraries(${COMPONENT_LIB} INTERFACE "-u _Z5setupv -u _Z4loopv")
```

Comment:
How do you know what is the proper mangle prefix/suffix? Asking because we might need to add some more functions to the list

Reply:
You can run something like:
```bash
echo "void setup() {}" | g++ -c -x c++ -o - - | nm -
```
Output:
```
0000000000000000 T __Z5setupv
```